addons/product/models/product.py

```diff
@@ -345,6 +345,16 @@ def create(self, vals_list):
                 product._set_standard_price(vals.get('standard_price') or 0.0)
         # `_get_variant_id_for_combination` depends on existing variants
         self.clear_caches()
+        self.env['product.template'].invalidate_cache(
+            fnames=[
```

Comment:
Not confortable with this, what is this doing? Maybe @rdeodoo should give his input on the whole commit?

Reply:
That's the correct way to invalidate the cache. Codewise looks fine.
In this case I guess you want the following fields to be recomputed (they are all computed field) as the newly created product will/might change those field values.

As for the whole commit I am not confortable with deprecating methods in stable but as the whole code as already been refactored anyway..

Just triple check that it won't break anything after pulling this change on existing DB and then after updating modules.

Reply:
Deprecating is fine to me, actually this is in stable where it makes the more sense since in master we would rather delete than deprecate.

Reply:
Of course, I meant deprecated _that much_ methods by that :yum:

Reply:
@rdeodoo correctly explained for the cache, the ORM does not invalidate it when records are created, so the mentioned fields won't have the correct value after the create if they had been cached previously.

This is not a big deal in general because usually records are created in a different request/env than the one they are read (so there is no cache issue), but this is typically an issue on tests where everything is done on the same env.

I have done the cache invalidation in that commit: 5b32f1b5cdbe71301896b251b7a0a4b81601a46e
and back then I forgot to add this one, but the test I added here highlighted the issue.